app/models/task.rb

```diff
@@ -41,13 +42,19 @@ def accept_notes
     def update_job
       @job = self.job
       @job.update(done: @job.tasks.select{|x| x.status != "accepted"}.empty?)
-      if @job.done? && @job.assignment.awards.count < 3 && @job.assignment.assignment_type != "test"
-        @job.awards.create(rank: @job.assignment.awards.count + 1)
-
+      if @job.done?
+        if @job.assignment.awards.count < 3 && @job.assignment.assignment_type != "test"
+          @job.awards.create(rank: @job.assignment.awards.count + 1)
+        end
         assignment = @job.assignment
         if assignment.jobs.where(done: false).empty?
           assignment.update(done: true)
         end
+      else
+        @job.assignment.update(done: false)
+        if @job.awards.count > 0
+          @job.awards.first.destroy
```

Comment:
I'm speaking about the second case. After you remove an award from a job, you probably need to update other awards too.
Example: assignment has 3 awards and you're deleting the first, a gold one. Will you leave other awards as they are after that?